### DISCUSSION TITLE
Disabling Base64 Encoding for User Metadata

### DISCUSSION THREAD

**Discussion by khakem** [zitadel/zitadel#8836] [General]
Hello,
I am configuring OAuth2 authentication with ZITADEL and Grafana, and I would like to know if it is possible to disable Base64 encoding for user metadata. Currently, the metadata is encoded in Base64 within the JWT token, which poses challenges for extracting and using this data in Grafana.

Could you please provide instructions or indicate if there is a configuration parameter to store and transmit user metadata without Base64 encoding? Any help or documentation on this matter would be greatly appreciated.

Best regards

**Comment by hifabienne** [comment]
Hi @khakem 
At the moment we always encode the data. 
As we don't know what is stored and also want to give you the possibility to store whatever you like, it is encode.

**Comment by khakem** [comment]
Thank you for your response and your responsiveness.
I am currently configuring OAuth2 authentication with ZITADEL and Grafana. Our goal is to map user roles and organizational attributes efficiently within Grafana. However, we are facing a challenge with the Base64 encoding of user metadata.
Use Case:
We need to extract organizational attributes (e.g., orgId) from the JWT token provided by ZITADEL.
These attributes are currently encoded in Base64, which complicates the extraction and utilization within Grafana.
Questions:
Is there a way to configure ZITADEL to avoid Base64 encoding of user metadata?
If not, do you have any recommendations for handling this scenario efficiently within Grafana or through a middleware?
Your assistance in resolving this issue would be greatly appreciated.
Best regards

**Reply by hifabienne** [reply]
There is no possibility to configure to avoid the base64 encoding.
As the implementation would need to be implemented on grafana side and i am not sure what possibilities you have there, I can't really give you a recommendation
